### PR TITLE
Plugin host

### PR DESCRIPTION
**Pull request by Xcelled**
Won't work fully till https://github.com/timeyyy/system_hotkey/issues/7 is fixed, but it's a start.

### CODE REVIEW

File: hotkeys.py [hunk 1]
```diff
@@ -44,47 +44,89 @@ def load(self):
...
 		if name in self.commands:
 			log.warning('Registering over existing command name %s', name)
 		#endif
 
 		self.commands[name] = function
+
+		log.debug('Registered command {} for {}'.format(name, function))
+	#enddef
+
+	# TODO: These should directly use QKeySequence in the config
```
Comment (Xcelled) [inline hotkeys.py:93]: @logicplace This should at least be order-agnostic. Ideally we'd use keyseq directly instead of its string representation.

Comment (logicplace) [reply hotkeys.py:93]: You can do that with the logging function too. Using {} doesn't make it order-agnostic. log.debug('... %(name)s ...', name=string)

Comment (Xcelled) [reply hotkeys.py:93]: ??? I'm taking about the hotkey sequence as a key in the config dict, which was referenced by your TODO that I replaced. Did github not  link to the right line again?

Converting the hotkey seq to a QKeySeq and then that to a string *should* normalize the modifier order, but I'd like to directly use QKeySequences as keys in the config dict. All that's missing is for the config to parse from strings on load, and re-stringify on save.

Comment (logicplace) [reply hotkeys.py:93]: o idk ok http://screencloud.net/v/gG4c

Comment (Xcelled) [reply hotkeys.py:93]: I guess it did, but... confusingly? Idk. Anyway I'm not gonna screw with config.

**Comment by Xcelled** [comment]
The system hotkey issue has been fixed upstream.

**Pull request merged**